### PR TITLE
Splitting up tests and covering new method.

### LINKED ISSUE #6: separate tests by view
Separate the tests by views, rather than grouping them all together in a single test.

### LINKED ISSUE #9: Split test
Each assertion in test_feed_content should be in separate tests.

### PR DESCRIPTION
**Pull request by somexpert**
Should fix #6 and fix #9. Basically this branch is taking tests that had loops and splitting them into individual tests, as well as breaking up one of the larger tests with multiple assertions and adding a test for the model's is_success() method.

### CODE REVIEW

**Comment by somexpert** [comment]
@ldko, @damonkelley, this is ready for review. There will be more changes to the tests, per the remaining issues, so don't worry just yet that I'm not validating the xml against schemas. That will come soon.

**Comment by ldko** [comment]
Structurally would it make sense instead of having the `TestLogic` class to have that broken into three different classes, one for testing the HTTP responses, one for the correct views called, and one for the correct template used?

**Comment by ldko** [comment]
Also there is the nitpicky thing about docstrings and how url is an acronym: URL.

**Comment by somexpert** [comment]
I can do that. I was struggling to find a proper title for that class anyways. So how does this sound:
-  class TestURLs
- class TestViewsCalled
- class TestTemplatesUsed
- class TestLogic

That last class might need a different name. It would have the last 6 tests that are currently in that class. Or should those be broken down into two smaller classes as well?

And yeah, I didn't mean to re-introduce the capitalization errors, I'll fix that right up.
What other docstring concerns do you have?

**Comment by damonkelley** [comment]
IMO, the name "TestLogic" is too vague. What logic are we testing? I agree with Lauren's suggestion.

File: tests/tests.py [hunk 1]
```diff
@@ -29,55 +29,105 @@ def create_event(title='test', outcome='Success', name='John Doe'):
...
+        url = reverse('major-event-log:feed')
+        self.assertEqual(self.client.get(url).status_code, 200)
+
+    def test_about_url(self):
+        """Check that the about url receives an HTTP 200."""
+        url = reverse('major-event-log:about')
+        self.assertEqual(self.client.get(url).status_code, 200)
+
+    def test_event_details_view_called(self):
+        """Check that the event_details page calls the correct view."""
```
Comment (damonkelley) [inline tests/tests.py]: I think we need to replace "page" with a different word. 

Perhaps "Check that the event_details URL pattern calls the correct view."

**Comment by damonkelley** [comment]
"TestContentProduced" is very vague as well.

**Comment by somexpert** [comment]
How about breaking up "TestLogic" into two smaller classes? They could be titled "TestModelMethods" and "TestGetEventOr404". Additionally, the "TestContentProduced" class could be changed to "TestXML" or something similar, though some of the tests may have to be removed from that class that don't deal directly with xml.

File: tests/tests.py (hunk 1 above)
Comment (somexpert) [inline tests/tests.py]: Yeah, I agree that "page" should be replaced. I'll make that change.

**Comment by damonkelley** [comment]
Do you mean after you pull out the url, view, and template tests? Thus `TestLogic` would become 5 classes altogether?

**Comment by somexpert** [comment]
Exactly

**Comment by damonkelley** [comment]
Did you consider leveraging the `setUp` method to do your event creation, rather than defining `create_event`. FWIW, `setUp` would allow you to assign the event to `self` and it would be easily accessible inside the test methods.

**Comment by somexpert** [comment]
I've briefly looked at it, and yeah, I think you're right, that would be better than the setup I have now, especially since the number of individual tests that need created events has just jumped up significantly.

**Comment by damonkelley** [comment]
I like the way you test the URLs. You use the client and verify the status code, _and_ you check the expect view is called. :100: 

However, I think we can improve on the testing of the view logic and templates. Instead of going "through" the url, check out using a the `RequestFactory` to create a dummy request object and send it directly to the view, instead letting the test client do it. This will make the tests more _unit_ and less _functional_. From looking through the tests really quick, it doesn't look like there is much "view logic" that is being tested (not that that is a bad thing, your views are not complicated at all) aside from which template is being delivered, and for those tests, we definitely don't _need_ to go through the URL.

Link to the RequestFactory docs https://docs.djangoproject.com/en/1.8/topics/testing/advanced/

**Comment by somexpert** [comment]
Ah, right, I remember you mentioning that several weeks ago, but it slipped my mind. It shall be done!

**Comment by somexpert** [comment]
On a related note, do you think I should use a similar approach for testing the content (by which I mean both the xml being generated and the plain text being displayed on the pages)?

**Comment by damonkelley** [comment]
Using a request factory?

**Comment by somexpert** [comment]
Yes

**Comment by damonkelley** [comment]
Yeah, if you are going to test content, I would go with the `RequestFactory`. No reason to involve the URLs if it is not necessary, especially since they are not the system under test.

**Comment by somexpert** [comment]
Will do.

**Comment by somexpert** [comment]
So I've been trying to replace my usage of the test client's get method with the request factory's get method, and I'm running into some issues due to the fact that they do not return the same response object. The test client's responses have very useful extra data added on which makes the tests much nicer IMHO. So I'm wondering if it's absolutely necessary to switch over. I'm showing the same test with both methods below. Note that this is a _best-case_ comparison. Other tests, such as those with class-based views, end up considerably worse when switching to the request factory.

```
response = self.client.get(reverse('major-event-log:event_atom', args=[self.events[0].id]))
self.assertEqual(response.context['event'], self.events[0])
```

vs

```
request = self.factory.get(reverse('major-event-log:event_details', args=[self.events[0].id]))
response = views.event_details(request, str(self.events[0].id))
self.assertIn(str(self.events[0].id), response.content)
```

Not too bad, but as I said, this was best-case, and there are still things I don't like about it. I don't think it looks as clean, and I don't like that I'm no longer able to assert from the context. Instead, I'm now doing a string search, which I personally think is something to be avoided when possible, as what is being displayed in the page could change at any moment. So what do you guys think?

**Comment by somexpert** [comment]
Essentially I think the decreased compartmentalization of using the test client is worth the following benefits:
1. The tests mimic actual usage more closely, going through the full path to get the rendered page
2. The tests are more flexible to changes in the templates
3. I think the performance is less variable, as a very large webpage would take longer to look for an occurrence of a substring than simply checking the context would

**Comment by damonkelley** [comment]
Regarding your first comment,

Don't worry about switching if it means that we will need to test for a string **in** the template.

When it comes to the complexity added when using the request factory, keep in mind that it removes the need to go through the urls. The point of it is to give us a request object that we can pass directly to the view. Thus, there is no need to call reverse to get a path. I tend to do something like this

``` python
def my_test(self):
    # The path '/' is meaningless in this context.
    request = self.factory.get('/')
    response = views.my_view(request)
    # make assertion on response
```

Regarding your second comment,
2. I am with you 100% here :wink:

**Comment by somexpert** [comment]
Well I'll see if that leaves any tests that are prime candidates for switching over. I think there may be none that fit the bill. Except maybe the current xml tests, which I don't want to touch because they're going to be completely changed in the next branch.

On an unrelated note, should I just delete the `test_index_events()`? Seems that since it's a class based view that I have not written any custom methods or functions for that it's essentially testing django rather than my own code. Any thoughts?

**Comment by somexpert** [comment]
I think the same case could be made for `test_feed_events()`.

**Comment by ldko** [comment]
I think if you are leaving testing your views to go through the URL, it is ok to leave tests in for those URLs to .

**Comment by somexpert** [comment]
Well in that case, @ldko, @damonkelley , this PR is ready for review.

**Comment by damonkelley** [comment]
@somexpert, my GH handle is @damonkelley :wink:

**Comment by somexpert** [comment]
right, I knew that........

**Comment by ldko** [comment]
:+1:

**Comment by damonkelley** [comment]
:+1:

**Pull request merged**